core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -278,29 +281,8 @@ public ServerStreamListener streamCreated(final ServerStream stream, final Strin
       // Run in serializingExecutor so jumpListener.setListener() is called before any callbacks
       // are delivered, including any errors. Callbacks can still be triggered, but they will be
       // queued.
-      serializingExecutor.execute(new Runnable() {
-          @Override
-          public void run() {
-            ServerStreamListener listener = NOOP_LISTENER;
-            try {
-              ServerMethodDefinition<?, ?> method = registry.lookupMethod(methodName);
-              if (method == null) {
-                stream.close(
-                    Status.UNIMPLEMENTED.withDescription("Method not found: " + methodName),
-                    new Metadata());
-                timeout.cancel(true);
-                return;
-              }
-              listener = startCall(stream, methodName, method, timeout, headers);
-            } catch (Throwable t) {
-              stream.close(Status.fromThrowable(t), new Metadata());
-              timeout.cancel(true);
-              throw Throwables.propagate(t);
-            } finally {
-              jumpListener.setListener(listener);
```

Comment:
The problem with this is that its harder to test.  The Runnable is created and run in a way that is inaccessible to a test.  It is really hard to tell when the listener is added (explicitly, and in a controlled way).  

Thoughts?